funcs_scripts.php

```diff
@@ -243,8 +262,14 @@ function is_docs()
  */
 function is_gha_repository()
 {
-    global $MODULE_DIR;
-    return !is_module() && strpos($MODULE_DIR, '/gha-') !== false;
+    global $GITHUB_REF;
+    return in_array(
+        $GITHUB_REF,
+        array_column(
+            MetaData::getAllRepositoryMetaData()[MetaData::CATEGORY_WORKFLOW],
+            'github'
+        )
+    );
```

Comment:
All gha-* repos are in the workflow category, and nothing else is.

Reply:
Should revert this, this one is like is_module() where we're not using MetaData as source of truth.  This function is specifically for gha repos, and the naming convention on gha- will always be there, though there is some chance someone will put in a non-gha workflow file in the workflow section of repositories.json.

Reply:
The reason we're not using `MetaData` in `is_module()` is because there might be repositories in other categories which are also modules, so we need to fall back on composer.json values.

This function is different. There's no need to fall back on anything, everything we need to know is in the `CATEGORY_WORKFLOW` category. All of the repositories in that category are the "gha-*" repositories. Nothing else is in that category. This is robust, and is much better than checking against a naming convention.

> there is some chance someone will put in a non-gha workflow file in the workflow section of repositories.json.

If they do that, they're doing the wrong thing.
The same can be said of naming conventions - there is a chance someone could make a workflow repository that doesn't follow the naming convention, or could make another repository that accidentally starts with "gha-".

Both of those are a low chance and aren't worth considering IMO.

Reply:
For context, the whole reason that category even exists in the file is so you can tell if something's a gha repo by just checking if it's in that category.